camp/apps/monitors/purpleair/tasks.py

```diff
@@ -16,12 +17,13 @@
 from camp.apps.monitors.purpleair.models import PurpleAir
 
 
-@db_periodic_task(crontab(minute='*/2'), priority=50)
+@db_periodic_task(crontab(minute='*/1'), priority=50)
 def import_recent_data():
     print('[import_recent_data]')
-    for monitor in PurpleAir.objects.all():
-        print('\n' * 10, '[import_recent_data]', monitor.name, '\n' * 10)
-        import_monitor_data.schedule([monitor.pk], delay=1, priority=30)
+    if HUEY.pending_count() < settings.MAX_QUEUE_SIZE:
+        for monitor in PurpleAir.objects.all():
```

Comment:
Since this task is being updated to run every minutes, I want to try and avoid hitting the PurpleAir API needlessly. One option would be to filter this query by monitors that haven't been updated in the last 2 minutes. If `PurpleAir.latest` were a `ForeignKey` field, we could do something like...

```
PurpleAir.objects.filter(latest__timestamp__lte=timezone.now() - timedelta(minutes=2))
```

But it's a `JSONField`, so... I wonder if that would work? I'm guessing the datetime passed in might get converted to a string, which is still sortable (but not (currently) indexed, so it wouldn't be super efficient – though with only a couple hundred PurpleAir's, that may not matter much).

Reply:
Hmm. I don't know. Worth thinking about some more, I suppose.